crates/miniserve/src/lib.rs

```diff
@@ -32,15 +27,26 @@ pub enum Content {
 pub type Response = Result<Content, http::StatusCode>;
 
 /// Trait alias for functions that can handle requests and return responses.
-pub trait Handler: Fn(Request) -> Response + Send + Sync + 'static {}
+pub trait Handler: Fn(Request) -> Self::Future + Send + Sync + 'static {
+    type Future: Future<Output = Response> + Send + Sync + 'static;
+}
```

Comment:
Take a close look at the change to the `Handler` trait. Previously, a handler was a function that took a request and returned a response. Now, a handler is a function that takes a request and returns a *future* which eventually returns a response.